crates/ruff_python_ast/src/name.rs

```diff
@@ -1,9 +1,213 @@
+use std::borrow::{Borrow, Cow};
 use std::fmt::{Debug, Display, Formatter, Write};
 use std::hash::{Hash, Hasher};
 use std::ops::Deref;
 
 use crate::{nodes, Expr};
 
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize, ruff_macros::CacheKey,)
+)]
+pub struct Name(compact_str::CompactString);
+
+impl Name {
+    #[inline]
+    pub fn empty() -> Self {
+        Self(compact_str::CompactString::default())
+    }
+
+    #[inline]
+    pub fn new(name: impl AsRef<str>) -> Self {
+        Self(compact_str::CompactString::new(name))
+    }
+
+    #[inline]
+    pub fn new_static(name: &'static str) -> Self {
+        // TODO: Use ComactString::const_new once we upgrade to 0.8 https://github.com/ParkMyCar/compact_str/pull/336
```

Comment:
Nit: can you add your name to the TODO? (Also: typo in `ComactString`)

Reply:
Sure, what's the benefit of doing it? I'm not necessarily the person fixing it and the git history already shows who added the TODO.